clients/web/src/presentation/pages/auth/Signup.tsx

```diff
@@ -0,0 +1,157 @@
+import React, { useEffect, useState } from 'react';
+import { Link, useNavigate } from 'react-router-dom';
+
+import authService from '@shared/services/authService';
+import useAuthenticatedUser from '@web/hooks/useAuthenticatedUser';
+import Button from '@web/presentation/components/button';
+import InputField from '@web/presentation/components/input-fields';
+import PasswordField from '@web/presentation/components/input-fields/PasswordInput';
+import { Toast } from '@web/presentation/components/toast';
+import { toastHideDisappearTime } from '@web/constants/common';
+import { DashboardPath, LoginPath } from '@web/constants/routeConsts';
+import { validatePassword } from '@web/utils/validationUtils';
+import { MdOutlineMail, HiOutlinePhone } from '@web/presentation/assets/icons';
+
+const SignUp: React.FC = () => {
+  const navigate = useNavigate();
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [organizationName, setOrganizationName] = useState('');
+  const [phoneNumber, setPhoneNumber] = useState('+88');
+  const [toastVisible, setToastVisible] = useState(false);
+  const [toastMsg, setToastMsg] = useState('');
+  const [toastClass, setToastClass] = useState('');
+  const { user } = useAuthenticatedUser();
+  const [loading, setLoading] = useState(false);
+
+  const handleError = (error: string) => {
+    setToastMsg(error);
+    setToastClass('alert-error');
+    setToastVisible(true);
+
+    setTimeout(() => {
+      setToastVisible(false);
+    }, toastHideDisappearTime);
+  };
+
+  const handleSignUp = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    const validationError = validatePassword(password, confirmPassword);
+    if (validationError != null) {
+      handleError(validationError);
+      return;
+    }
+
+    try {
+      setLoading(true);
```

Comment:
this is something that we will need everywhere, I feel like the loading, data, error syntax should have a reusable hook
I've used something here
https://github.com/sakibstark11/preact-stuff/blob/main/src/hooks/useFetchData.ts

but you can also use other existing tools.

Reply:
Added reusable hook for fetching data 5004fc30f12951f78bcdba7270fe66ee3b1341b7